test_cache.py fails on ubuntu linux with django 1.7

Discovered this while using tox, the following combination consistently fails on master:

```
msql version: 5.6.27-0ubuntu1
branch: master
Python 2.7.8
Django version: 1.7.10.final.0
```

Tested this on my personal and work laptop. Numerous tests in test_cache.py fail with the following message:

```
self = <tests.testapp.test_cache.MySQLCacheTests testMethod=test_keys_with_prefix_version_4_custom_key2>

    @wraps(func)
    def wrapper(self):
>       return func(self, *args, **kwargs)

.tox/py27-django17/local/lib/python2.7/site-packages/ddt.py:146: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/testapp/test_cache.py:1113: in test_keys_with_prefix_version
    cache.set('V12', True, version=1)
django_mysql/cache.py:204: in set
    self._base_set('set', key, value, timeout)
django_mysql/cache.py:228: in _base_set
    cursor.execute(query.format(table=table), params)
.tox/py27-django17/local/lib/python2.7/site-packages/django/db/backends/utils.py:65: in execute
    return self.cursor.execute(sql, params)
django_mysql/monkey_patches.py:42: in execute
    return orig_execute(self, sql, args)
.tox/py27-django17/local/lib/python2.7/site-packages/django/db/backends/mysql/base.py:129: in execute
    return self.cursor.execute(query, args)
.tox/py27-django17/local/lib/python2.7/site-packages/MySQLdb/cursors.py:222: in execute
    if not self._defer_warnings: self._warning_check()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MySQLdb.cursors.Cursor object at 0x7fd2c1531950>

    def _warning_check(self):
        from warnings import warn
        if self._warnings:
            warnings = self._get_db().show_warnings()
            if warnings:
                # This is done in two loops in case
                # Warnings are set to raise exceptions.
                for w in warnings:
                    self.messages.append((self.Warning, w))
                for w in warnings:
>                   warn(w[-1], self.Warning, 3)
E                   Warning: Invalid utf8mb4 character string: '800288'
```
the utf8mb4 character string value changes between each test, but the error is basically the same.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,6 @@
 sudo: required
+dist: precise
+group: edge
 
 notifications:
   email: false

--- a/.travis/before_script.sh
+++ b/.travis/before_script.sh
@@ -1,7 +1,10 @@
 #!/bin/bash -e
 
+set -o verbose
+
 # Add percona - have to use non-apt version since Travis' ubuntu 12.04 repo is
 # way out of date
+sudo apt-get update
 sudo apt-get install -y libio-socket-ssl-perl
 wget https://www.percona.com/downloads/percona-toolkit/2.2.13/deb/percona-toolkit_2.2.13_all.deb
 sudo dpkg -i percona-toolkit_2.2.13_all.deb

--- a/tests/settings.py
+++ b/tests/settings.py
@@ -4,7 +4,7 @@ import os
 import django
 
 BASE_DIR = os.path.dirname(os.path.dirname(__file__))
-DEBUG = True
+DEBUG = False
 TEMPLATE_DEBUG = DEBUG
 
 SECRET_KEY = 'THISuISdNOT9A$SECRET9x&ji!vceayg+wwt472!bgs$0!i3k4'

--- a/tests/testapp/test_cache.py
+++ b/tests/testapp/test_cache.py
@@ -4,7 +4,6 @@ from __future__ import unicode_literals
 import imp
 import os
 import time
-import warnings
 from decimal import Decimal
 from unittest import skipUnless
 
@@ -896,13 +895,10 @@ class MySQLCacheTests(MySQLCacheTableMixin, TestCase):
         cache.key_func = func
 
         try:
-            with warnings.catch_warnings(record=True) as w:
-                warnings.simplefilter("always")
+            with pytest.warns(CacheKeyWarning):
                 # memcached does not allow whitespace or control characters in
                 # keys
-                cache.set('key with spaces', 'value')
-                assert len(w) == 2
-                assert isinstance(w[0].message, CacheKeyWarning)
+                cache.set('space key', 'value')
 
             with pytest.raises(ValueError):
                 # memcached limits key length to 250

